doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,621 @@
+
+If you try to add edges to vertices with invalid IDs (i.e., you try to add an edge to vertex `5` when the graph has only three vertices), you get an `igraph.InternalError` exception:
+
+```{r echo = TRUE, eval = FALSE}
+gi <- add_edges(gi, edges = c(5,4))
+```
+
+The message tries to explain what went wrong (`cannot add edges. -- Invalid vertex id`).
+
+Let us add some more vertices and edges to our graph:
+
+```{r echo = TRUE}
+gi <- add_edges(gi, edges = c(3,1))
+gi <- add_vertices(gi, 3)
+gi <- add_edges(gi, edges = c(3,4, 4,5, 5,6, 6,4))
+print(gi)
+```
+
+We now have an directed graph with 6 vertices and 7 edges. Vertex and edge IDs are always *continuous*, so if you delete a vertex all subsequent vertices will be renumbered. When a vertex is renumbered, edges are **not** renumbered, but their source and target vertices will. Use `Graph.delete_vertices` and `Graph.delete_edges` to perform these operations. For instance, to delete the edge connecting vertices `3-4`, get its ID and then delete it:
+
+```{r echo = TRUE}
+get.edge.ids(gi, c(3,4))
+```
+
+```{r}
+gi <- delete.edges(gi, 4)
+```
+
+## Generating graphs
+
+igraph includes both deterministic and stochastic graph generators (see `generation`). *Deterministic* generators produce the same graph every time you call the fuction, e.g.:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- make_tree(127, 2, mode = "undirected")
+summary(gi)
+```
+
+Uses `make_tree` to generate a regular tree graph with 127 vertices, each vertex having two children (and one parent, of course). No matter how many times you call `make_tree`, the generated graph will always be the same if you use the same parameters:
+
+```{r}
+gi2 <- make_tree(127, 2, mode = "undirected")
+```
+
+```{r echo = TRUE}
+get.edgelist(gi) == get.edgelist(gi2)
+```
+
+The above code snippet also shows you that the `get_edgelist()` method, which returns a list of source and target vertices for all edges, sorted by edge ID. If you print the first 10 elements, you get:
+
+```{r echo = TRUE}
+head(get.edgelist(gi2),10)
+```
+
+*Stochastic* generators produce a different graph each time, e.g. `sample_grg`:
+
+```{r echo = TRUE}
+gi <- make_empty_graph()
+gi <- sample_grg(100, 0.2)
+summary(gi)
+```
+
+This generates a geometric random graph: *n* points are chosen randomly and uniformly inside the unit square and pairs of points closer to each other than a predefined distance *d* are connected by an edge. If you generate GRGs with the same parameters, they will be different:
+
+```{r echo = TRUE}
+gi2 <- make_empty_graph()
+gi2 <- sample_grg(100, 0.2)
+
+gi_a <- head(get.edgelist(gi), 10) 
+gi2_a <-head(get.edgelist(gi2), 10)
+
+gi_a == gi2_a
+```
+
+A slightly looser way to check if the graphs are equivalent is via `isomorphic()`:
```

Comment:
Can you explain to readers what it means for graphs to be isomorphic, and how that differs from the check you did above, as well as from `identical_graphs`?